Lib/enum.py

```diff
@@ -1177,7 +1177,8 @@ def _generate_next_value_(name, start, count, last_values):
         count: the number of existing members
         last_value: the last value assigned or None
         """
-        for last_value in reversed(last_values):
+        numerical_last_values =  [x for x in last_values if isinstance(x, int) or isinstance(x, float)]
```

Comment:
Please remove this line -- it unnecessarily excludes any data type that can add 1 to itself:

    >>> from fractions import Fraction
    >>> Fraction(2, 3) + 1
    Fraction(5, 3)

    >>> from decimal import Decimal
    >>> Decimal('179.333') + 1
    Decimal('180.333')

I'm sure there are others.

Reply:
That's a good point, but then how do we deal with `TypeError: '<' not supported between instances of 'str' and 'int'` when trying to use `sorted(last_values, reverse=True)`? Any suggestions?

Reply:
Could potentially mimic the behavior you had? e.g.
```python 
for i in range(1, len(last_values)):
    try:
        srt = sorted(last_values[:len(last_values)+1-i], reverse=True)
        break
    except TypeError:
        continue
```